### PR TITLE
Standardize autoHighlight logic in CompositeLayer

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Background

This PR addresses an issue where `autoHighlight` on a composite layer only highlights the picked sublayer. For example, in PolygonLayer, GeoJsonLayer etc., hovering over a polygon can only highlight either the outline or the fill, while we would expect the whole feature to be highlighted. This also applies to custom composite layers (e.g. a "NodeLayer" that draws a circle behind a text label) where multiple sublayers share the same data.

#### Change List
- Add `updateAutoHighlight` method to `Layer`
- When the hovered object changes, by default update the highlight on every sublayer inside the top-level layer, instead of just the picked one
- Implement special highlighting logic in GeoJsonLayer, TileLayer and MVTLayer
- Tests

### CODE REVIEW

File: modules/core/src/lib/layer.js [hunk 1]
```diff
@@ -864,6 +864,26 @@ export default class Layer extends Component {
     }
   }
 
+  updateAutoHighlight(info) {
```
Comment (ibgreen) [inline modules/core/src/lib/layer.js:867]: Isn't this the new "lifecycle method" that should be documented and overridden by subclasses?

File: modules/core/src/lib/layer.js [hunk 2]
```diff
@@ -864,6 +864,26 @@ export default class Layer extends Component {
...
+    if (this.props.autoHighlight) {
+      this._updateAutoHighlight(info);
+    }
+  }
+
+  // May be overriden by classes
+  _updateAutoHighlight(info) {
```
Comment (ibgreen) [inline modules/core/src/lib/layer.js:874]: Nit: I have to admit that I find this pair of functions with and without underscore somewhat confusing. I wonder if there could be a naming scheme that makes things clearer. 

I think at some point the private method naming across the `Layer` class could stand a review. I have heard external users complain about it being hard to follow the logic.

File: modules/core/src/lib/composite-layer.js [hunk 3]
```diff
@@ -55,10 +55,11 @@ export default class CompositeLayer extends Layer {
   // override Layer.getPickingInfo() because decoding / setting uniform do
   // not apply to a composite layer.
   // @return null to cancel event
-  getPickingInfo({info}) {
+  getPickingInfo({info, sourceLayer}) {
```
Comment (ibgreen) [inline modules/core/src/lib/composite-layer.js:58]: Nit: I assume we can't just add the sourceLayer to `info` from the caller?

This is a public `Layer` "lifecycle method"? Then we should probably document it.

**Review by ibgreen — APPROVED** [review]
Looks good. Added some thoughts.

File: modules/core/src/lib/layer.js (hunk 2 above)
Comment (Pessimistress) [reply modules/core/src/lib/layer.js:874]: We have several pairs like this in the Layer class, e.g. `updateAttributes` and `_updateAttributes`. I agree that a review would help clarify things.

File: modules/core/src/lib/layer.js (hunk 1 above)
Comment (Pessimistress) [reply modules/core/src/lib/layer.js:867]: `updateAutoHighlight` is meant to be called by code outside of this class, and `_updateAutoHighlight` is meant to be overriden by sub classes.

I prefer not to document this until we have a consensus on the right naming.

File: modules/core/src/lib/composite-layer.js (hunk 3 above)
Comment (Pessimistress) [reply modules/core/src/lib/composite-layer.js:58]: https://deck.gl/docs/api-reference/core/composite-layer#getpickinginfo

File: modules/core/src/lib/layer.js (hunk 2 above)
Comment (alasarr) [reply modules/core/src/lib/layer.js:874]: This PR looks great to me in terms of architecture. I've also faced a couple of times with the issue it tries to solve. I've to say I felt the same confusion with the naming of `_updateAutoHighlight`and `updateAutoHighlight`.

However, I've to admit I find nothing better
